babel/dates.py

```diff
@@ -1282,6 +1284,16 @@ def format_timezone(self, char, num):
                 return get_timezone_name(self.value.tzinfo, width,
                                          uncommon=True, locale=self.locale)
             return get_timezone_location(self.value.tzinfo, locale=self.locale)
+        # Included additional elif condition to add support for 'X' in timezone format
+        elif char in "xX":
+            if get_timezone_gmt(self.value, width='short', locale=self.locale)=='+0000' and char == 'X':
+                return 'Z'
+            elif num == 1:
+                return get_timezone_gmt(self.value, width='short', locale=self.locale)[:-2]
```

Comment:
Yes I was working on that, but I have one concern in function get_timezone_gmt. When we use width='short' it is returning some thing like '-0700' where as it should return 'GMT-7'. Is this bug?